### PR TITLE
Magiclysm: Spell encumbrance and minor spellcasting ui tweaks

### PR DESCRIPTION
**Pull request by KorGgenT**
#### Summary

```SUMMARY: Mods "Add penalties to spellcasting while encumbered based on the spell"```
#### Purpose of change

It was a little too powerful to be able to wear whatever you wannted and be able to cast all of your spells at full power.
#### Describe the solution

See these flags: 
- SOMATIC, // arm encumbrance affects fail % and casting time (slightly)
- NO_HANDS, // hands do not affect spell energy cost
- NO_LEGS, // legs do not affect casting time
- CONCENTRATE, // focus affects spell fail %
image
image

As you can see, I added the spell class at the top, and enumerated some information about the spell: the flags shown above, plus whether the spell goes through walls (only if the spell can make use of it, i.e. range > 1)
I also fixed the bug that showed a healing message if you do 0 damage to yourself with a spell.

### CODE REVIEW

File: src/magic.cpp [hunk 1]
```diff
@@ -1045,6 +1088,59 @@ static std::string moves_to_string( const int moves )
...
+{
+    if( !sp.has_flag( spell_flag::NO_HANDS ) ) {
+        return std::max( 0, p.encumb( bp_hand_l ) + p.encumb( bp_hand_r ) - 10 ) > 0;
+    }
+    return false;
+}
+
+// this prints various things about the spell out in a list
+// including flags and things like "goes through walls"
+std::string enumerate_spell_data( const spell &sp )
```
Comment (ZhilkinSerg) [inline src/magic.cpp]:
```suggestion
static std::string enumerate_spell_data( const spell &sp )
```

**Comment by Temptdos** [comment]
This is probably not the right place for it, but just wanted to confirm:

As it stands now, wearing only scavenger gear with 20 arm encumbrance, Nature's Bow cannot be cast as it's too difficult and spells like magic missile and wooden shaft take a 40% reduction in DPS.

**Comment by KorGgenT** [comment]
i'm probably going to let the balance shake out a little before i adjust it again. this is mostly a rough estimation of what i want with implementation, for now. as far as nature's bow not being able to be cast, try leveling it up to decrease its spell failure % or increasing your spellcraft skill or intelligence.

**Comment by Temptdos** [comment]
Sounds good! I'm currently at 16 intelligence (dump stat at character creation + mutations) but my Nature's Bow is admittedly only level 1. Also, can you only increase your spellcraft skill via character creation or is there another method you need to find in the game? It's only level 3 from character creation.

Otherwise is it planned for dexterity/perception to reduce the encumbrance penalty for casting in the future?

**Comment by KorGgenT** [comment]
spellcraft increases by studying spells in spellbooks. capped at the spell's difficulty.

**Comment by Temptdos** [comment]
Awesome, and thanks for making this mod!

**Comment by DorfGuy1** [comment]
Love it, reminds me of DnD wherein wizards can't wear heavy armor because they need unristricted movement.

**Pull request merged**